plotMDTS_function.R

```diff
@@ -100,6 +100,43 @@ plotMDTS <- function(
   max.val <- max(data.group$value, na.rm = TRUE)
   colnames(data.group)[colnames(data.group)=="value"] <- metric
   
+  # Add growth calculation
+  if (growth) {
+    
+    # Only by.date = "day" is currently supported
+    if (!by.date == "day") {
+      stop("Growth calculations are only supported wtih by.date = 'Day'")
+    }
+    
+    data.group <- addGrowthRate(data.group,
+                                metric = metric,
+                                date.col = "date",
+                                growth.per = growth.per)
+    
+    metric = "growth.rate"
+  }
+  
+  # Add time shift
+  if (!is.null(time.shift)) {
+    
+    # TODO (mbh): Figure out what to shift by depending on date aggregation?
+    data.group <- addTimeShift(data.group, 
+                               metric = metric,
+                               time.shift = time.shift, 
+                               shift.per = shift.per)
+  }
+  
+  # Aggregate by date
+  data.group <- aggregateByDate(as.data.frame(data.group), by.date = by.date)
```

Comment:
I'm confused -- doesn't aggregating by date make the previous growth rate calculations meaningless?